test/connection-manager/index.spec.ts

```diff
@@ -169,7 +169,72 @@ describe('Connection Manager', () => {
     expect(shortestLivedWithLowestTagSpy).to.have.property('callCount', 1)
   })
 
-  it('should close connection when the maximum has been reached even without tags', async () => {
+  it('should not close connection that is on the allowlist when pruning', async () => {
+    const max = 5
```

Comment:
I think this test could be simplified by setting the max to 1, two peers - one with a remote address in the allow list and one not and asserting that the one not in the allow list was disconnected.

Reply:
I think it's still best to give all peers tags and then have the lowest tagged peer in the `allowlist`. Since normally the _lowest tag value_  is pruned first I wanted to highlight that this would not be the case given **that said peer** is in the `allowlist`